build/management/commands/build_citations.py

```diff
@@ -44,32 +45,41 @@ def create_citations(self):
 			refs = yaml.load(refs_yaml, Loader=yaml.FullLoader)
 		wr = WebResource.objects.get(slug='doi')
 		pubjournal = None
+
 		# Create main publication first for empty publication cells for non-published tools
 		for url, vals in refs.items():
-			pubjournal = PublicationJournal.objects.get_or_create(slug='-'.join(vals['Journal'].split(' ')).lower(), name=vals['Journal'])[0]
 			if vals['Default']=='GPCRdb':
-				main_gpcrdb_pub = self.create_publication(vals['DOI'], wr, pubjournal)
+				main_gpcrdb_pub = Publication.get_or_create_from_doi(vals['DOI'])
 			if vals['Default']=='GproteinDb':
-				main_gproteindb_pub = self.create_publication(vals['DOI'], wr, pubjournal)
+				main_gproteindb_pub = Publication.get_or_create_from_doi(vals['DOI'])
+
 		for url, vals in refs.items():
 			doi = vals['DOI']
+			pub = False
 			if vals['Journal'] in ['Preprint at Research Square', 'Submitted']:
-				pubjournal = PublicationJournal.objects.get(name=vals['Journal'])
-			pub = self.create_publication(doi, wr, pubjournal)
+				pubjournal, created = PublicationJournal.objects.get_or_create(name__iexact=vals['Journal'],
+					defaults={'slug': slugify(vals['Journal'])})
+				pub = self.create_publication(doi, wr, pubjournal)
```

Comment:
you kept this here because these 2 exceptions would work with the new function?

Reply:
Indeed, for these two functions there wouldn't be any automatic handler, so this would be the way to go. The preprints are indexed now at PMID, so that could work but then we would lose your annotation "Preprint at Research square". So just keeping it.